### PR TITLE
Support service generation on action folders

### PR DESCRIPTION
**Pull request by hidmic**
Connected to ros2/rosidl#301. This pull request is part of a series of pull requests that make the necessary changes to allow us to build .srv files outside an `srv` folder.

connects to ros2/rosidl#301

### CODE REVIEW

File: rosidl_generator_dds_idl/cmake/rosidl_generate_dds_interfaces.cmake [hunk 1]
```diff
@@ -54,6 +55,8 @@ macro(rosidl_generate_dds_interfaces target)
         list(APPEND _generated_msg_files "${_output_path}/${_name}_.idl")
       elseif(_parent_folder STREQUAL "srv")
         list(APPEND _generated_srv_files "${_output_path}/${_name}_.idl")
+      elseif(_parent_folder STREQUAL "action")
```
Comment (jacobperron) [inline rosidl_generator_dds_idl/cmake/rosidl_generate_dds_interfaces.cmake]: Do we also want another `elseif` block in the parent scope for the file extension? And maybe more importantly if a .srv file appears inside an action folder? E.g.

```cmake
elseif(_extension STREQUAL ".srv")
  if(_parent_folder STREQUAL "srv")
    list(APPEND _generated_srv_files "${_output_path}/Sample_${_name}_Request_.idl")
    list(APPEND _generated_srv_files "${_output_path}/Sample_${_name}_Response_.idl")
  elseif(_parent_folder STREQUAL "action")
    list(APPEND _generated_action_files "${_output_path}/Sample_${_name}_Request_.idl")
    list(APPEND _generated_action_files "${_output_path}/Sample_${_name}_Response_.idl")
  else()
    ...
  endif()
elseif(_extension STREQUAL ".action")
  # Do something here?
else()
  ...
endif()
```

**Comment by jacobperron** [comment]
LGTM

**Review by jacobperron — APPROVED** [review]
Pending CI

**Review by sloretz — APPROVED** [review]

**Comment by apojomovsky** [comment]
Ok, given that we have green light I'm proceeding to merge this and its related PRs

**Pull request merged**